source/plugin/data/customdata.rst

```diff
@@ -0,0 +1,297 @@
+players last attacker, therefore we will have only the data of the last attackers UUID.
+
+.. code-block:: java
+
+    import java.util.UUID;
+
+    public class LastAttackerDataSerilizable {
+    
+        private UUID lastAttackerId;
+    
+    }
+
+.. note::
+
+    Any data you wish to store must be able to be serialized into java primitives and/or strings
+
+
+From here you will want to implement the javadoc:`DataSerializable` which will give you two methods to implement. The
+first is ``contentVersion`` which is for the version of your data manipulator. The other method (``toContainer``) is 
+used for serializing your data to the dataholder it belongs to. To do this you need to create a new :javadoc:`DataContainer`
+then set your value(s) to the newly created ``DataContainer``
+
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.persistence.DataSerializable;
+    import org.spongepowered.api.data.persistence.DataContainer;
+    import org.spongepowered.api.data.persistence.DataQuery;
+
+    public class LastAttackerDataSerilizable implements DataSerializable {
+    
+        private UUID lastAttackerId;
+
+        public static final DataQuery UUID_PATH = DataQuery.of("attack", "last");
+
+        @Override
+        public int contentVersion(){
+            return 1;
+        }
+
+        @Override
+        public DataContainer toContainer(){
+            return DataContainer.createNew()
+                .set(LastAttackerDataSerilizable.UUID_PATH, lastAttackerId.toString());
+        }
+    
+    }    
+
+After that you will want to create a class that can build the data from a ``DataContainer`` this is known as 
+the :javadoc:`DataBuilder` which can be implemented as follows:
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.persistence.InvalidDataException;
+
+    public class LastAttackerDataBuilder implements DataBuilder<LastAttackerDataSerilizable> {
+    
+        @Override
+        public Optional<LastAttackerDataSerilizable> build(DataView container) throws InvalidDataException {
+            Optional<String> lastAttackerAsStringId container.getString(LastAttackerDataSerilizable.UUID_PATH);
+            if(lastAttackerAsStringId.isPresent()){
+                UUID lastAttacker = UUID.fromString(lastAttackerAsStringId.get());
+                return Optional.of(new LastAttackerDataSerilizable(lastAttacker));
+            }
+            return Optional.empty();
+        }
+    
+    }
+
+
+Registration
+============
+
+Registering your ``DataSerializable`` allows it to be accessible by Sponge and by other plugins in a generic way. The
+game/plugin can create copies of your data and serialize/deserialize your data without referencing any of your classes
+directly.
+
+To register a ``DataSerializable`` Sponge has the :javadoc:`RegisterDataEvent` event. This will allow you to register
+your data with the appropriate ``DataHolder``
```

Comment:
I'd almost reword this that the `DataSerializable` isn't the core intent of it being accessible, moreso that it's persisted to Minecraft's traditional data objects that can support custom data (Items, Entities, BlockEntities), accessible via the author's registered `Key`.

Reply:
Specifically mentioning that custom data is most easily accessible via `Key`s, `DataSerializable` is only one of two options to making data persisted.